airflow_adapter.py

```diff
@@ -0,0 +1,119 @@
+"""The Airflow Adapter."""
+from adapter_abc import Adapter
+from filter_releases import filter_releases
+from filter_releases import sort
+from from_sql import read_releases
+from from_sql import read_tasks
+from to_sql import to_sql_release
+from to_sql import to_sql_releases
+from to_sql import to_sql_task
+from to_sql import to_sql_tasks
+
+
+class AirflowAdapter(Adapter):
+  """Provides a way to interact with the Airflow Database, and get data from it."""
+
+  def __init__(self, airflow_db):
+    self._airflow_db = airflow_db
+
+    self._releases = self._airflow_db.query('SELECT * FROM dag_run;')
```

Comment:
why do we need this ?

Reply:
It is there so that we can assemble branch and type information at the start, rather than doing it for each and every request. I suppose we could move these to get_branches() and get_types, however.

Reply:
I understand this, but those values will change as we create more releases, and therefore instead of doing it in the constructor you should do it in a different methods that you can call when your cache is invalidated.